### PR TITLE
 Connect to and poll Mqtt broker

### PR DESCRIPTION
**Pull request by hdoordt**

### CODE REVIEW

**Review by tdittr — APPROVED** [review]
Looks good do far, but I feel like we should look at our overall structure soon. Maybe write some documentation as it gets harder and harder to follow.

File: firmware/src/adc_capture.rs [hunk 1]
```diff
@@ -25,65 +28,89 @@ struct DoubleBufferedRingBuffer {
...
                     Ordering::Equal =>  (RingRange::Normal(0..afternoon.end), morning.start..self.buffer.len()),
                     Ordering::Greater => unreachable!("morning.start was already bigger the length or the length as not divisible by CONVS_PER_CHUNK"),
                 }
-            },
+            }
         };
 
         self.free_range = new_range;
-        let dma_buffer = &mut self.buffer[dma_range].try_into().expect("dma_range is always CONVS_PER_CHUNK long");
+        todo!()
```
[the remark this answers is not in the document]
Comment (tdittr) [reply firmware/src/adc_capture.rs:62]: Oh I see now that adc_capture is still a WIP, so thi sis fine

File: firmware/src/main.rs [hunk 2]
```diff
@@ -47,13 +48,10 @@ defmt::timestamp!("{=u64:iso8601ms}", {
 
 #[app(device = stm32f7xx_hal::pac, dispatchers = [CAN1_RX0])]
 mod app {
-    use defmt::Debug2Format;
-    use static_cell::StaticCell;
-    use stm32_eth::ptp::{EthernetPTP, Subseconds, Timestamp};
-    use stm32f7xx_hal::pac;
+
+    use minimq::Publication;
```
[the remark this answers is not in the document]
Comment (tdittr) [reply firmware/src/main.rs:52]: Looking at it again this can be resolved in the next round...

**Comment by hdoordt** [comment]
@tdittr yeah totally

**Pull request merged**